src/App.tsx

```diff
@@ -1,21 +1,25 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, Suspense, lazy } from 'react'
 import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom'
 import { useWallet } from '@binance-chain/bsc-use-wallet'
 import { ResetCSS } from '@pancakeswap-libs/uikit'
 import BigNumber from 'bignumber.js'
 import { useFetchPublicData } from 'state/hooks'
 import GlobalStyle from './style/Global'
 import Menu from './components/Menu'
-import Farms from './views/Farms'
-import Farm from './views/Farm'
-import Home from './views/Home'
-import Lottery from './views/Lottery'
-import Pools from './views/Pools'
-import Ifos from './views/Ifos'
-import NotFound from './views/NotFound'
-import Nft from './views/Nft'
+import PageLoader from './components/PageLoader'
 import NftGlobalNotification from './views/Nft/components/NftGlobalNotification'
 
+// Route-based code splitting
+const Home = lazy(() => import('./views/Home'))
+const Farms = lazy(() => import('./views/Farms'))
+const Farm = lazy(() => import('./views/Farm'))
+const Lottery = lazy(() => import('./views/Lottery'))
+const Pools = lazy(() => import('./views/Pools'))
+const Ifos = lazy(() => import('./views/Ifos'))
+const NotFound = lazy(() => import('./views/NotFound'))
```

Comment:
Probably not needed for this but we should consider keeping critical routes in the bundle.